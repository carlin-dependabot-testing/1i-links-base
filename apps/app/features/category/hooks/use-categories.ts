import { useMutation, useQuery, useQueryClient } from '@tanstack/react-query'

import { calculateLinkCounts } from '@/utils/calculate-link-counts'

import { getCategoriesPath, getCategoryApiRoute } from '@/constants/routes'
import { useLinks } from '@/features/links/hooks/links/use-links'
import { type InferCategorySchema } from '@/features/links/schemas/category.schema'
import { type Category } from '@/features/links/types/category.types'

import { useToast } from '@links-base/ui/hooks'

export function useCategories() {
  const { toast } = useToast()
  const queryClient = useQueryClient()
  const { links, isLoading: linksLoading } = useLinks()

  // Main categories query
  const {
    data: categories = [] as InferCategorySchema[],
    isLoading: categoriesLoading,
    error,
    refetch: refreshCategories
  } = useQuery({
    queryKey: ['categories', links?.length],
    queryFn: async () => {
      const response = await fetch(getCategoriesPath(), {
        cache: 'no-store',
        headers: {
          'Cache-Control': 'no-cache'
        }
      })
      const categoriesData = await response.json()

      if (!Array.isArray(categoriesData)) {
        throw new Error('Invalid categories data')
      }

      // Wait for links to be loaded
      if (linksLoading || !links) {
        return categoriesData.map((category: Category) => ({
          ...category,
          linkCount: 0
        }))
      }

      const linkCounts = calculateLinkCounts(categoriesData, links)

      return categoriesData.map((category: Category) => ({
        ...category,
        linkCount: linkCounts.get(category.name.toLowerCase()) || 0
      })) as InferCategorySchema[]
    },
    enabled: true
  })

  // Calculate uncategorized categories
  const uncategorizedCategories = new Set(
    !linksLoading && links
      ? links
          .map(link => link.category.toLowerCase())
          .filter(
            category =>
              !categories.some(cat => cat.name.toLowerCase() === category)
          )
      : []
  )

  // Add category mutation
  const addCategoryMutation = useMutation({
    mutationFn: async (category: Omit<Category, 'id'>) => {
      const response = await fetch(getCategoryApiRoute(), {
        method: 'POST',
        headers: { 'Content-Type': 'application/json' },
        body: JSON.stringify(category)
      })
      if (!response.ok) throw new Error('Failed to add category')
      return response.json()
    },
    onSuccess: () => {
      void queryClient.invalidateQueries({ queryKey: ['categories'] })
      toast({
        title: 'Category added successfully',
        description: 'The category has been added to the database.'
      })
    },
    onError: error => {
      console.error('Error adding category:', error)
      toast({
        title: 'Failed to add category',
        description:
          error instanceof Error ? error.message : 'An error occurred',
        variant: 'destructive'
      })
    }
  })

  // Edit category mutation
  const editCategoryMutation = useMutation({
    mutationFn: async ({
      oldName,
      newCategory
    }: {
      oldName: string
      newCategory: InferCategorySchema
    }) => {
      const response = await fetch(getCategoryApiRoute(), {
        method: 'PUT',
        headers: { 'Content-Type': 'application/json' },
        body: JSON.stringify({
          oldName,
          newName: newCategory.name,
          icon: newCategory.icon,
          description: newCategory.description
        })
      })
      if (!response.ok) throw new Error('Failed to edit category')
    },
    onSuccess: () => {
      void queryClient.invalidateQueries({ queryKey: ['categories'] })

      toast({
        title: 'Category updated successfully',
        description: 'The category has been updated in the database.'
      })
    },
    onError: () => {
      toast({
        title: 'Failed to update category',
        description:
          'There was an error updating the category in the database.',
        variant: 'destructive'
      })
    }
  })

  // Delete category mutation
  const deleteCategoryMutation = useMutation({
    mutationFn: async ({
      categoryName,
      deleteLinks
    }: {
      categoryName: string
      deleteLinks: boolean
    }) => {
      const response = await fetch(getCategoryApiRoute(), {
        method: 'DELETE',
        headers: { 'Content-Type': 'application/json' },
        body: JSON.stringify({ categoryName })
      })
      if (!response.ok) throw new Error('Failed to delete category')
    },
    onSuccess: () => {
      void queryClient.invalidateQueries({ queryKey: ['categories'] })
      toast({
        title: 'Category deleted successfully',
        description: 'The category has been removed from the database.'
      })
    },
    onError: error => {
      console.error('Error deleting category:', error)
      toast({
        title: 'Failed to delete category',
        description:
          error instanceof Error ? error.message : 'An error occurred',
        variant: 'destructive'
      })
    }
  })

  const isLoading = linksLoading || categoriesLoading

  return {
    categories,
    uncategorizedCategories,
    isLoading,
    error,
    refreshCategories,
    editCategory: editCategoryMutation.mutateAsync,
    addCategory: addCategoryMutation.mutateAsync,
    deleteCategory: deleteCategoryMutation.mutateAsync
  }
}
