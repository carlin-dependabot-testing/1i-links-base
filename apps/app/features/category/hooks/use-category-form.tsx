import { useForm } from 'react-hook-form'

import { zodResolver } from '@hookform/resolvers/zod'

import {
  CategorySchema,
  type InferCategorySchema
} from '@/features/links/schemas/category.schema'

export const useCategoryForm = (
  initialValues: Partial<InferCategorySchema> = {},
  onSubmit: (data: InferCategorySchema) => Promise<void>
) => {
  const form = useForm<InferCategorySchema>({
    resolver: zodResolver(CategorySchema),
    defaultValues: {
      name: initialValues.name ?? '',
      icon: initialValues.icon ?? 'Folder'
    }
  })

  const handleSubmit = async (data: InferCategorySchema) => {
    try {
      await onSubmit(data)
      form.reset()
      return true
    } catch (error) {
      console.error('Failed to submit category:', error)
      return false
    }
  }

  return {
    form,
    handleSubmit: form.handleSubmit(handleSubmit),
    isSubmitting: form.formState.isSubmitting
  }
}
