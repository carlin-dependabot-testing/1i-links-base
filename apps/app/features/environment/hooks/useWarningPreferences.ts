import { useLocalStorage } from 'usehooks-ts'

import { STORAGE_KEYS } from '@/constants/storage'

export type WarningKey =
  | typeof STORAGE_KEYS.HAS_SEEN_ENVIRONMENT_WARNING
  | typeof STORAGE_KEYS.HAS_SEEN_HIDDEN_CATEGORIES_ALERT

/**
 * Hook for managing warning preferences
 */
export function useWarningPreferences() {
  const [hasSeenEnvironmentWarning, setHasSeenEnvironmentWarning] =
    useLocalStorage(STORAGE_KEYS.HAS_SEEN_ENVIRONMENT_WARNING, false)
  const [hasSeenHiddenCategoriesAlert, setHasSeenHiddenCategoriesAlert] =
    useLocalStorage(STORAGE_KEYS.HAS_SEEN_HIDDEN_CATEGORIES_ALERT, false)

  const resetAllWarnings = () => {
    setHasSeenEnvironmentWarning(false)
    setHasSeenHiddenCategoriesAlert(false)
  }

  return {
    hasSeenEnvironmentWarning,
    setHasSeenEnvironmentWarning,
    hasSeenHiddenCategoriesAlert,
    setHasSeenHiddenCategoriesAlert,
    resetAllWarnings
  }
}
