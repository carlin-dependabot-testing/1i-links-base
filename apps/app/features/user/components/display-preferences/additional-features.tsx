'use client'

import { Label } from '@links-base/ui/label'
import { Switch } from '@links-base/ui/switch'

export type AdditionalFeaturesProps = {
  showFavorites: boolean
  onShowFavoritesChange: (checked: boolean) => void
}

export const AdditionalFeatures = ({
  showFavorites,
  onShowFavoritesChange
}: AdditionalFeaturesProps) => {
  return (
    <div className="space-y-4">
      <div className="text-sm font-medium">Additional Features</div>
      <div className="grid gap-4">
        <div className="grid grid-cols-3 items-center gap-4">
          <Label htmlFor="show-favorites">Show Favorites</Label>
          <div className="col-span-2">
            <Switch
              id="show-favorites"
              checked={showFavorites}
              onCheckedChange={onShowFavoritesChange}
            />
          </div>
        </div>
      </div>
    </div>
  )
}
