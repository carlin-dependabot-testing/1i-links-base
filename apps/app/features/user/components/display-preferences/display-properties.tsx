'use client'

import { Button } from '@links-base/ui/button'
import { cn } from '@links-base/ui/utils'

type DisplayProperty = {
  id: string
  label: string
  isEnabled: boolean
  onToggle: (value: boolean) => void
}

type DisplayPropertiesProps = {
  properties: DisplayProperty[]
}

export const DisplayProperties = ({ properties }: DisplayPropertiesProps) => {
  if (properties.length === 0) return null

  return (
    <div className="space-y-4">
      <div className="text-sm font-medium" id="display-properties-heading">
        Display Properties
      </div>
      <div
        className="flex flex-wrap gap-2"
        role="group"
        aria-labelledby="display-properties-heading"
      >
        {properties.map(({ id, label, isEnabled, onToggle }) => (
          <Button
            key={id}
            variant={isEnabled ? 'default' : 'outline'}
            className={cn(
              'h-7 rounded-full px-3 text-xs',
              isEnabled ? 'hover:bg-primary/80' : 'hover:bg-muted'
            )}
            onClick={() => onToggle(!isEnabled)}
            role="switch"
            id={`display-property-${id}`}
          >
            <span className="sr-only">Show </span>
            {label}
            <span className="sr-only">
              {isEnabled ? ' (enabled)' : ' (disabled)'}
            </span>
          </Button>
        ))}
      </div>
    </div>
  )
}
