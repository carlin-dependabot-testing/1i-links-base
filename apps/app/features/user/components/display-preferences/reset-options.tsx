'use client'

import { RefreshCw, RotateCcw } from 'lucide-react'
import { toast } from 'sonner'

type ResetOptionsProps = {
  onResetWarnings: () => void
  onResetDefaults: () => void
}

export const ResetOptions = ({
  onResetWarnings,
  onResetDefaults
}: ResetOptionsProps) => {
  return (
    <div className="flex flex-col gap-2">
      <div
        className="flex cursor-pointer items-center gap-2 text-sm hover:text-primary"
        onClick={() => {
          onResetWarnings()
          toast.success('All warning dialogs have been reset')
        }}
      >
        <RotateCcw className="h-4 w-4" />
        Reset Warning Dialogs
      </div>

      <div
        className="flex cursor-pointer items-center gap-2 text-sm hover:text-primary"
        onClick={() => {
          onResetDefaults()
          toast.success('Display preferences restored to defaults')
        }}
      >
        <RefreshCw className="h-4 w-4" />
        Reset to Default Settings
      </div>
    </div>
  )
}
