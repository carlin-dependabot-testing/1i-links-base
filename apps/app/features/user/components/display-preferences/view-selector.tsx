'use client'

import { Label } from '@links-base/ui/label'
import { RadioGroup, RadioGroupItem } from '@links-base/ui/radio-group'
import {
  Tooltip,
  TooltipContent,
  TooltipProvider,
  TooltipTrigger
} from '@links-base/ui/tooltip'
import { cn } from '@links-base/ui/utils'

import { type CardSize } from '../../stores/useUserSettingsStore'

type ViewSelectorProps = {
  cardSize: CardSize
  onViewChange: (value: CardSize) => void
}

export const ViewSelector = ({ cardSize, onViewChange }: ViewSelectorProps) => {
  return (
    <div className="grid gap-2">
      <Label>View</Label>
      <RadioGroup
        value={cardSize}
        onValueChange={onViewChange}
        className="grid grid-cols-2 gap-2"
      >
        <TooltipProvider>
          <Tooltip>
            <TooltipTrigger asChild>
              <div>
                <RadioGroupItem
                  value="compact"
                  id="compact"
                  className="peer sr-only"
                />
                <Label
                  htmlFor="compact"
                  className={cn(
                    'flex cursor-pointer flex-col items-center justify-between rounded-md border-2 border-muted bg-popover p-4 transition-colors hover:bg-accent hover:text-accent-foreground',
                    cardSize === 'compact' && 'border-primary'
                  )}
                >
                  <div className="grid w-full gap-1 text-center">
                    <div className="space-y-2">
                      <div className="mx-auto h-6 w-8/12 rounded-md bg-muted" />
                    </div>
                  </div>
                  <span className="mt-2 block w-full text-center text-sm font-normal">
                    Compact
                  </span>
                </Label>
              </div>
            </TooltipTrigger>
            <TooltipContent>
              <p>Compact view with title only</p>
            </TooltipContent>
          </Tooltip>

          <Tooltip>
            <TooltipTrigger asChild>
              <div>
                <RadioGroupItem
                  value="default"
                  id="default"
                  className="peer sr-only"
                />
                <Label
                  htmlFor="default"
                  className={cn(
                    'flex cursor-pointer flex-col items-center justify-between rounded-md border-2 border-muted bg-popover p-4 transition-colors hover:bg-accent hover:text-accent-foreground',
                    cardSize === 'default' && 'border-primary'
                  )}
                >
                  <div className="grid w-full gap-1">
                    <div className="space-y-2">
                      <div className="h-2 w-full rounded-md bg-muted" />
                      <div className="h-1 w-full rounded-md bg-muted" />
                    </div>
                  </div>
                  <span className="mt-2 block w-full text-center text-sm font-normal">
                    Default
                  </span>
                </Label>
              </div>
            </TooltipTrigger>
            <TooltipContent>
              <p>Default view with description and tags</p>
            </TooltipContent>
          </Tooltip>
        </TooltipProvider>
      </RadioGroup>
    </div>
  )
}
