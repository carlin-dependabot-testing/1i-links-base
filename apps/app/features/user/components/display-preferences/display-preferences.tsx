'use client'

import { Settings2 } from 'lucide-react'

import { useWarningPreferences } from '@/features/environment/hooks/useWarningPreferences'
import {
  type CardSize,
  useNonDefaultSettings,
  useUserSettingsStore
} from '@/features/user/stores/useUserSettingsStore'

import { Button } from '@links-base/ui/button'
import { Popover, PopoverContent, PopoverTrigger } from '@links-base/ui/popover'
import { Separator } from '@links-base/ui/separator'

import { AdditionalFeatures } from './additional-features'
import { DisplayProperties } from './display-properties'
import { ResetOptions } from './reset-options'
import { ViewSelector } from './view-selector'

export const DisplayPreferences = () => {
  const {
    cardSize,
    showGradients,
    showFavorites,
    showTags,
    showDescriptions,
    showIcons,
    showEnvironmentBadges,
    setCardSize,
    setShowGradients,
    setShowFavorites,
    setShowTags,
    setShowDescriptions,
    setShowIcons,
    setShowEnvironmentBadges,
    resetToDefaults,
    previousDescriptionState,
    previousTagsState
  } = useUserSettingsStore()
  const { resetAllWarnings } = useWarningPreferences()
  const hasNonDefaultSettings = useNonDefaultSettings()

  const handleGradientToggle = (checked: boolean) => {
    setShowGradients(checked)
  }

  const handleViewChange = (value: CardSize) => {
    setCardSize(value)

    if (value === 'default') {
      setShowDescriptions(previousDescriptionState)
      setShowTags(previousTagsState)
    }

    if (value === 'compact') {
      setShowDescriptions(false)
      setShowTags(false)
    }
  }

  const displayProperties = [
    {
      id: 'icon',
      label: 'Icon',
      isEnabled: showIcons,
      onToggle: setShowIcons
    },
    {
      id: 'environment',
      label: 'Environment',
      isEnabled: showEnvironmentBadges,
      onToggle: setShowEnvironmentBadges
    },
    ...(cardSize === 'default'
      ? [
          {
            id: 'description',
            label: 'Description',
            isEnabled: showDescriptions,
            onToggle: setShowDescriptions
          },
          {
            id: 'tags',
            label: 'Tags',
            isEnabled: showTags,
            onToggle: setShowTags
          }
        ]
      : []),
    {
      id: 'gradients',
      label: 'Gradients',
      isEnabled: showGradients,
      onToggle: handleGradientToggle
    }
  ]

  return (
    <Popover>
      <PopoverTrigger asChild>
        <Button variant="outline" size="icon" className="relative">
          {hasNonDefaultSettings && (
            <span className="absolute right-0 top-0 flex h-2 w-2 -translate-y-1/4 translate-x-1/4">
              <span
                className="absolute inline-flex h-full w-full animate-ping rounded-full bg-blue-400 opacity-75"
                data-testid="settings-indicator"
              ></span>
              <span className="relative inline-flex h-2 w-2 rounded-full bg-blue-500"></span>
            </span>
          )}
          <Settings2 className="h-5 w-5" />
        </Button>
      </PopoverTrigger>
      <PopoverContent className="w-80">
        <div className="grid gap-4">
          <div className="space-y-2">
            <h4 className="font-medium leading-none">Display Preferences</h4>
            <p className="text-sm text-muted-foreground">
              Customize how your dashboard looks
            </p>
          </div>
          <Separator />
          <div className="space-y-4">
            <div className="text-sm font-medium">Layout</div>
            <ViewSelector cardSize={cardSize} onViewChange={handleViewChange} />
          </div>
          <Separator />
          <DisplayProperties properties={displayProperties} />
          <Separator />
          <AdditionalFeatures
            showFavorites={showFavorites}
            onShowFavoritesChange={setShowFavorites}
          />
          <Separator />
          <ResetOptions
            onResetWarnings={resetAllWarnings}
            onResetDefaults={resetToDefaults}
          />
        </div>
      </PopoverContent>
    </Popover>
  )
}
