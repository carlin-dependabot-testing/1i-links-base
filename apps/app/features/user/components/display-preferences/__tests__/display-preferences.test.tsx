import { fireEvent, render, screen } from '@/utils/test-utils'

import { useWarningPreferences } from '@/features/environment/hooks/useWarningPreferences'
import {
  useNonDefaultSettings,
  useUserSettingsStore
} from '@/features/user/stores/useUserSettingsStore'

import { DisplayPreferences } from '../display-preferences'

// Mock the hooks
jest.mock('@/features/user/stores/useUserSettingsStore')
jest.mock('@/features/environment/hooks/useWarningPreferences')

const renderComponent = () => {
  return render(<DisplayPreferences />)
}

describe('DisplayPreferences', () => {
  const mockUseUserSettingsStore = useUserSettingsStore as unknown as jest.Mock
  const mockUseWarningPreferences =
    useWarningPreferences as unknown as jest.Mock

  beforeEach(() => {
    // Reset all mocks before each test
    mockUseUserSettingsStore.mockReturnValue({
      cardSize: 'default',
      showGradients: true,
      showFavorites: true,
      showTags: true,
      showDescriptions: true,
      showIcons: true,
      showEnvironmentBadges: true,
      setCardSize: jest.fn(),
      setShowGradients: jest.fn(),
      setShowFavorites: jest.fn(),
      setShowTags: jest.fn(),
      setShowDescriptions: jest.fn(),
      setShowIcons: jest.fn(),
      setShowEnvironmentBadges: jest.fn(),
      resetToDefaults: jest.fn(),
      previousDescriptionState: true,
      previousTagsState: true
    })

    mockUseWarningPreferences.mockReturnValue({
      resetAllWarnings: jest.fn()
    })
  })

  it('should render DisplayPreferences correctly', () => {
    const { container } = renderComponent()
    expect(container).toMatchSnapshot()
  })

  it('should open popover when settings button is clicked', () => {
    renderComponent()

    const settingsButton = screen.getByRole('button')
    fireEvent.click(settingsButton)

    expect(screen.getByText('Display Preferences')).toBeInTheDocument()
    expect(
      screen.getByText('Customize how your dashboard looks')
    ).toBeInTheDocument()
  })

  it('should show non-default settings indicator when settings are changed', () => {
    const mockUseNonDefaultSettings = useNonDefaultSettings as jest.Mock
    mockUseNonDefaultSettings.mockReturnValue(true)

    renderComponent()

    // Check if the blue dot indicator is present
    expect(screen.getByTestId('settings-indicator')).toHaveClass('animate-ping')
  })

  it('should handle view change correctly', () => {
    const mockSetCardSize = jest.fn()
    const mockSetShowDescriptions = jest.fn()
    const mockSetShowTags = jest.fn()

    mockUseUserSettingsStore.mockReturnValue({
      ...mockUseUserSettingsStore(),
      setCardSize: mockSetCardSize,
      setShowDescriptions: mockSetShowDescriptions,
      setShowTags: mockSetShowTags
    })

    renderComponent()

    // Open popover
    fireEvent.click(screen.getByRole('button'))

    // Click compact view button
    fireEvent.click(screen.getByText('Compact'))

    expect(mockSetCardSize).toHaveBeenCalledWith('compact')
    expect(mockSetShowDescriptions).toHaveBeenCalledWith(false)
    expect(mockSetShowTags).toHaveBeenCalledWith(false)
  })

  it('should handle property toggles correctly', () => {
    const mockSetShowGradients = jest.fn()

    mockUseUserSettingsStore.mockReturnValue({
      ...mockUseUserSettingsStore(),
      setShowGradients: mockSetShowGradients
    })

    renderComponent()

    // Open popover
    fireEvent.click(screen.getByRole('button'))

    // Try different queries to find the element
    const gradientToggle =
      screen.getByRole('switch', { name: /gradients/i }) ||
      screen.getByLabelText(/gradients/i) ||
      screen.getByText(/gradients/i)

    fireEvent.click(gradientToggle)

    expect(mockSetShowGradients).toHaveBeenCalled()
  })

  it('should handle reset options correctly', () => {
    const mockResetToDefaults = jest.fn()
    const mockResetAllWarnings = jest.fn()

    mockUseUserSettingsStore.mockReturnValue({
      ...mockUseUserSettingsStore(),
      resetToDefaults: mockResetToDefaults
    })

    mockUseWarningPreferences.mockReturnValue({
      resetAllWarnings: mockResetAllWarnings
    })

    renderComponent()

    // Open popover
    fireEvent.click(screen.getByRole('button'))

    // Click reset buttons
    fireEvent.click(screen.getByText('Reset to Default Settings'))
    fireEvent.click(screen.getByText('Reset Warning Dialogs'))

    expect(mockResetToDefaults).toHaveBeenCalled()
    expect(mockResetAllWarnings).toHaveBeenCalled()
  })

  it('should restore previous description and tags state when switching back to default view', () => {
    const mockSetShowDescriptions = jest.fn()
    const mockSetShowTags = jest.fn()

    mockUseUserSettingsStore.mockReturnValue({
      ...mockUseUserSettingsStore(),
      cardSize: 'compact',
      previousDescriptionState: true,
      previousTagsState: false,
      setShowDescriptions: mockSetShowDescriptions,
      setShowTags: mockSetShowTags
    })

    renderComponent()

    // Open popover
    fireEvent.click(screen.getByRole('button'))

    // Find and click the default view option
    fireEvent.click(screen.getByLabelText(/default/i))

    // Verify previous states are restored
    expect(mockSetShowDescriptions).toHaveBeenCalledWith(true)
    expect(mockSetShowTags).toHaveBeenCalledWith(false)
  })
})
