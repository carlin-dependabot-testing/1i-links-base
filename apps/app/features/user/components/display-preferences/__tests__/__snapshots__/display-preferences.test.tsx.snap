// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayPreferences should render DisplayPreferences correctly 1`] = `
<div>
  <div
    class="group/sidebar-wrapper flex min-h-svh w-full has-[[data-variant=inset]]:bg-sidebar"
    style="--sidebar-width: 16rem; --sidebar-width-icon: 3rem;"
  >
    <script
      nonce=""
    >
      ((e,s,n,m,o,l,u,h)=&gt;{let c=document.documentElement,T=["light","dark"];function p(d){(Array.isArray(e)?e:[e]).forEach(g=&gt;{let k=g==="class",S=k&&l?o.map(f=&gt;l[f]||f):o;k?(c.classList.remove(...S),c.classList.add(d)):c.setAttribute(g,d)}),C(d)}function C(d){h&&T.includes(d)&&(c.style.colorScheme=d)}function a(){return window.matchMedia("(prefers-color-scheme: dark)").matches?"dark":"light"}if(m)p(m);else try{let d=localStorage.getItem(s)||n,g=u&&d==="system"?a():d;p(g)}catch(d){}})("class","theme","system",null,["light","dark"],null,true,true)
    </script>
    <button
      aria-controls="radix-:r0:"
      aria-expanded="false"
      aria-haspopup="dialog"
      class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-10 w-10 relative"
      data-state="closed"
      type="button"
    >
      <svg
        class="lucide lucide-settings2 h-5 w-5"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M20 7h-9"
        />
        <path
          d="M14 17H5"
        />
        <circle
          cx="17"
          cy="17"
          r="3"
        />
        <circle
          cx="7"
          cy="7"
          r="3"
        />
      </svg>
    </button>
    <div
      class="fixed bottom-4 right-4 z-50 flex items-center gap-2 rounded-lg bg-background/80 p-2 shadow-sm backdrop-blur-sm"
    />
    <div
      aria-label="Notifications (F8)"
      role="region"
      style="pointer-events: none;"
      tabindex="-1"
    >
      <ol
        class="fixed top-0 z-[100] flex max-h-screen w-full flex-col-reverse p-4 sm:bottom-0 sm:right-0 sm:top-auto sm:flex-col md:max-w-[420px]"
        tabindex="-1"
      />
    </div>
  </div>
</div>
`;
